### PR TITLE
Handle LightGBM models with '=' in feature names

### LINKED ISSUE #281: Ill-formed LightGBM model file when the model is fitted from a pd.DataFrame having the character '=' inside any of column name.
When the LightGBMClassifier is fitted form a `pandas.DataFrame` having the character `=` inside any of column name the `treelite.Model.load('model.txt', model_format='lightgbm')` raise an `Ill-formed LightGBM model file` exception:

```
TreeliteError                             Traceback (most recent call last)
<ipython-input-36-db95c3aef74c> in <module>
     12     f.write(txt)
     13 
---> 14 treelite_model = treelite.Model.load('model.txt', model_format='lightgbm')

~/.local/lib/python3.8/site-packages/treelite/frontend.py in load(cls, filename, model_format)
    445         model_format = model_format.lower()
    446         if model_format == 'lightgbm':
--> 447             _check_call(_LIB.TreeliteLoadLightGBMModel(c_str(filename),
    448                                                        ctypes.byref(handle)))
    449         elif model_format == 'xgboost':

~/.local/lib/python3.8/site-packages/treelite/core.py in _check_call(ret)
     48     """
     49     if ret != 0:
---> 50         raise TreeliteError(_LIB.TreeliteGetLastError().decode('utf-8'))
     51 
     52 

TreeliteError: [11:02:36] /workspace/src/frontend/lightgbm.cc:276: Check failed: rest.empty(): Ill-formed LightGBM model file
Stack trace:
  [bt] (0) /home/sadasml/.local/lib/python3.8/site-packages/treelite/lib/libtreelite.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x4f) [0x7f998ac33d2f]
  [bt] (1) /home/sadasml/.local/lib/python3.8/site-packages/treelite/lib/libtreelite.so(+0x12a0fd) [0x7f998acd80fd]
  [bt] (2) /home/sadasml/.local/lib/python3.8/site-packages/treelite/lib/libtreelite.so(treelite::frontend::LoadLightGBMModel(char const*)+0x28) [0x7f998acdf598]
  [bt] (3) /home/sadasml/.local/lib/python3.8/site-packages/treelite/lib/libtreelite.so(TreeliteLoadLightGBMModel+0x16) [0x7f998ac2eee6]
  [bt] (4) /lib/x86_64-linux-gnu/libffi.so.7(+0x6ff5) [0x7f9ccfd1fff5]
  [bt] (5) /lib/x86_64-linux-gnu/libffi.so.7(+0x640a) [0x7f9ccfd1f40a]
  [bt] (6) /usr/lib/python3.8/lib-dynload/_ctypes.cpython-38-x86_64-linux-gnu.so(_ctypes_callproc+0x58c) [0x7f9ccedff29c]
  [bt] (7) /usr/lib/python3.8/lib-dynload/_ctypes.cpython-38-x86_64-linux-gnu.so(+0x137d0) [0x7f9ccedff7d0]
  [bt] (8) /usr/bin/python3(_PyObject_MakeTpCall+0x296) [0x5f66f6]
```

Manually editing the  `model.txt`  file to remove the symbol `=` solves the problem.

### PR DESCRIPTION
**Pull request by hcho3**
Closes #281

### CODE REVIEW

File: tests/python/test_lightgbm_integration.py [hunk 1]
```diff
@@ -152,6 +153,54 @@ def test_categorical_data(tmpdir, quantize, parallel_comp, toolchain):
...
+
+    rng = np.random.default_rng(seed=0)
+    n_row = 100
+    x0 = rng.integers(low=0, high=3, size=n_row)
+    x1 = rng.integers(low=0, high=5, size=n_row)
+    X = np.column_stack((x0, x1))
+    noise = rng.standard_normal(size=n_row) * 0.1
+    y = (x0 * 10 - 20) + (x1 - 2) + noise
+    df = pd.DataFrame({'x0': x0, 'x1': x1}).astype('category')
+    df_dummies = pd.get_dummies(df, prefix=['x0', 'x1'], prefix_sep='=')
```
Comment (hcho3) [inline tests/python/test_lightgbm_integration.py:187]: This test case creates a Pandas dataframe with '=' in the column names.

File: src/frontend/lightgbm.cc [hunk 2]
```diff
@@ -243,7 +243,10 @@ inline std::unique_ptr<treelite::Model> ParseStream(std::istream& fi) {
     std::getline(ss, key, '=');
     std::getline(ss, value, '=');
     std::getline(ss, rest);
-    TREELITE_CHECK(rest.empty()) << "Ill-formed LightGBM model file";
+    if (!rest.empty()) {
+      value += "=";
+      value += rest;
+    }
```
Comment (hcho3) [inline src/frontend/lightgbm.cc:249]: This change lets us handle the line
```
feature_names=x0=0 x0=1 x1=0 x1=1 x1=2
``` 
in model.txt.

**Comment by hcho3** [comment]
@trivialfis Can you take a quick look and see if the test case looks reasonable?

**Review by trivialfis — COMMENTED** [review]
The test looks good.  Maybe we can add a check to see if the generated model does have such feature names to be sure that we are testing the right thing.

**Pull request merged**